tcms/issuetracker/types.py

```diff
@@ -383,6 +384,87 @@ def report_issue_from_testcase(self, caserun):
 
         return url + '/issues/new?' + urlencode(args, True)
 
+class REDMINE(IssueTrackerType):
+    """
+        Support for Redmine. Requires:
+
+        :api_url: - the API URL for your Redmine instance
+        :api_password: - Redmine API Key
+
+        Additional control can be applied via the ``JIRA_OPTIONS`` configuration
+        setting (in ``product.py``). By default this setting is not provided and
+        the code uses ``jira.JIRA.DEFAULT_OPTIONS`` from the ``jira`` Python module!
+    """
+
+    def __init__(self, tracker):
+        super(REDMINE, self).__init__(tracker)
+
+        if not self.is_adding_testcase_to_issue_disabled():
+            self.rpc = redminelib.Redmine(
+                tracker.api_url,
+                key=self.tracker.api_password
+            )
+
+    def add_testcase_to_issue(self, testcases, issue):
+        for case in testcases:
+            redmine_integration.RedmineThread(self.rpc, case, issue).start()
+
+    def all_issues_link(self, ids):
+        if not self.tracker.base_url:
+            return None
+
+        if not self.tracker.base_url.endswith('/'):
+            self.tracker.base_url += '/'
+
+        return self.tracker.base_url + 'issues?utf8=✓&set_filter=1&sort=id%3Adesc&f%5B%5D=issue_id&op%5Bissue_id%5D=%3D&v%5Bissue_id%5D%5B%5D={0}&f%5B%5D=&c%5B%5D=tracker&c%5B%5D=status&c%5B%5D=priority&c%5B%5D=subject&c%5B%5D=assigned_to&c%5B%5D=updated_on&group_by=&t%5B%5D='.format('%2C'.join(ids))
+
+    def report_issue_from_testcase(self, caserun):
+        """
+            For the HTML API description see:
+            https://confluence.atlassian.com/display/JIRA050/Creating+Issues+via+direct+HTML+links
+        """
+
+        project = ""
+        projects = self.rpc.project.all()
+        for prj in projects:
+            if (prj.name==caserun.run.plan.product.name):
+                project = prj
+                break
+
+        if (project == ""):
```

Comment:
Use `if not project` and also fix the rest of the pylint & flake8 errors

Reply:
Fixed